bundles/org.eclipse.swt/Eclipse SWT Tests/win32/org/eclipse/swt/widgets/ControlWin32Tests.java

```diff
@@ -107,6 +110,27 @@ public void testCorrectScaleUpUsingDifferentSetBoundsMethod() {
 				new Rectangle(0, 82, 350, 83), button.getBoundsInPixels());
 	}
 
+	@ParameterizedTest
+	@CsvSource({ "0.5, 100, true", "1.0, 200, true", "2.0, 200, true", "2.0, quarter, true", "0.5, 100, false",
+			"1.0, 200, false", "2.0, 200, false", "2.0, quarter, false", })
+	public void autoScaleImageData0(float scaleFactor, String autoScale, boolean monitorSpecificScaling) {
+		DPIUtil.setMonitorSpecificScaling(monitorSpecificScaling);
+		DPIUtil.setAutoScaleValue(autoScale);
```

Comment:
Since the value is only initialized on class loading (i.e., at JVM startup), changing this value in a test without resetting it will leave the JVM in an unexpected state for the subsequent tests. To avoid that, the original value should be stored and reapplied after executing the test. This has to be done in the finally block, as otherwise a test failure would still leave behind an unexpected system state.